base/array.jl

```diff
@@ -649,7 +649,7 @@ end
 
 # returns the index of the next non-zero element, or 0 if all zeros
 function findnext(A, start::Integer)
-    for i = start:length(A)
+    for i = start:length(A) #Fixme iter
```

Comment:
This version of `findnext` explicitly works on linear indices, so I guess that's OK. We may want to add a version taking a `CartesianIndex` in the future. Same for the following methods.